Show article edit date in dashboard

jprask opened the pull request:
* Create edited? and readable_edit_date methods for article model
* Show article edit date in dashboard for edited articles
* Add some styling

<!--- Prepend PR title with [WIP] if work in progress. Remove when ready for review. -->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

Add article edited date in dashboard 

## Related Tickets & Documents

closes #3274

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Desktop:

![image](https://user-images.githubusercontent.com/28638133/60389808-17209680-9a9f-11e9-85b6-50ad7580a76b.png)

Mobile:

![image](https://user-images.githubusercontent.com/28638133/60389824-48996200-9a9f-11e9-870c-bcfd9225f1ba.png)


## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

![clock](https://media.giphy.com/media/tn8zWeNYA73G0/giphy.gif)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/thepracticaldev/dev.to?pullRequest=3325) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/thepracticaldev/dev.to?pullRequest=3325) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/thepracticaldev/dev.to?pullRequest=3325) it.</sub>

rhymes commented on app/models/article.rb:
```diff
@@ -374,6 +374,18 @@ def tag_keywords_for_search
+  end
+
+  def readable_edit_date
+    if edited_at && edited_at.year == Time.current.year
```
I think you can just quit this method so you don't have to check `edited_at`, nor use the `&.` operator.

```ruby
def readable_edit_date
  return unless edited_at
  ...
end
```

I would also add a test :-)

rhymes commented on app/views/dashboards/_dashboard_article.html.erb:
```diff
@@ -24,7 +24,14 @@
+        <span>
+          Published: <%= article.readable_publish_date %>
+        </span>
+        <span>
```
this is going to add an empty span if there's no edit date, I think the check should be done outside of it

jprask commented:
Whats up @rhymes, no problem! will consider you comments and push sutff soon

jprask commented:
Hey, pushed a version considering the review, putting it besides publish date. Also updated the screenshots. I personally prefer it [below](https://imgur.com/a/P5HLNq2), maybe with a few tweaks but up to you guys of course.

lightalloy commented on app/models/article.rb:
```diff
@@ -374,6 +374,20 @@ def tag_keywords_for_search
+    return unless edited?
+
+    if edited_at.year == Time.current.year
+      edited_at&.strftime("%b %e")
```
No need to use `&` since you have already returned when `edited_at` is `nil`

rhymes commented:
@jprask I've fixed the conflict, updated the code to use the `<time>` tag and replaced the comma with a dot, which looks better overall:

![Screenshot_2019-07-29 Dashboard - DEV(local) Community 👩‍💻👨‍💻](https://user-images.githubusercontent.com/146201/62042037-5ad10200-b1fd-11e9-9e17-fdfe0557be45.png)

lightalloy commented on app/models/article.rb:
```diff
@@ -374,6 +374,20 @@ def tag_keywords_for_search
+    edited_at.present?
+  end
+
+  def readable_edit_date
```
I think `readable_edit_date` fits more into decorator than a model. As I see, `article` in the `_dashboard_article` partial should be decorated already.
I would say the same about the `readable_publish_date`, but it needs additional checks on where it's used, and it's probably not the topic of this pr.

rhymes replied:
Since we didn't flag this initially, I didn't move it. Also I didn't want to change the original author's PR. But yeah, we might want to do this in this or another PR :)

rhymes replied:
@lightalloy if you agree we can move both in a decorator once this is merged

lightalloy replied:
@rhymes of course, that makes sense.

rhymes replied:
@lightalloy ok cool, I'm deferring the review to you because I've participated in the PR :D

lightalloy commented on app/views/dashboards/_dashboard_article.html.erb:
```diff
@@ -24,7 +24,16 @@
+      </span>
+
+      <% if article.edited? %>
+      <span>
```
A bit of indentation is missing :)

jprask commented:
> @jprask I've fixed the conflict, updated the code to use the `<time>` tag and replaced the comma with a dot, which looks better overall:
> 
> ![Screenshot_2019-07-29 Dashboard - DEV(local) Community 👩‍💻👨‍💻](https://user-images.githubusercontent.com/146201/62042037-5ad10200-b1fd-11e9-9e17-fdfe0557be45.png)

Looks good!
